docs/Developers/Profile_Bears.rst

```diff
@@ -0,0 +1,76 @@
+Profile Bears
+=============
+
+This document provides an overview of coala's Profiling Interface. The
+Profiling Interface provides abilities to profile Bear code to optimize its
+performance.
+
+The profiler will start by profiling the run() method of bears because this is
+the part where bear writers will spend time on, as rest of the part like loading
+the files, collecting the settings, etc. are done by coala itself.
+
+.. note::
+
+    Enabling both profiler and debugger on a bear at the same time will result
+    in error.
+
+Bear writers will have the ability to directly dump the raw profile output
+either on current working directory or to a specified directory name, which can
+be further used for examination of profiler stats with the help of different
+modules like ``pstats`` or ``snakeviz``.
+
+Usage
+-----
+
+coala's Profiler accepts an additional parameter: a directory path where
+profiled files will be dumped, if ``True`` is provided profiled files will be
+saved to current working directory. If the specified directory does not exist
+it will be created. If the specified path points to an already existing file
+a error is raised. Profiled files will get overwritten, if specified directory
```

Comment:
> Profiled files will get overwritten, if specified directory

I thought this was will cause an error?

If I am correct, you obviously have not re-checked this document and it shouldnt be up for review.

Reply:
`FileExistsError` will be raise when user provide a `filepath` in `--profile` but if user provide a `directory path` and that directory already have `profiled` files then in the next run of coala file with the same name gets overwritten